### DISCUSSION TITLE
Problems encrypting the "external" hard drive

### DISCUSSION THREAD

**Discussion by janivo** [szaimen/Nextcloud-NAS-Guide#99] [General]
Hey there,

thanks for your great guide.
I followed through until I had problems with [the encryption of the external hard drive](https://szaimen.github.io/Nextcloud-NAS-Guide/docs/format-encrypt-mount). The problem is that my hard drive is an internal M2.SDD which I cannot really connect/disconnect while running the system but which seems to be required in order to run the veracrypt BTFS script. I worked around this problem by hardcoding the reference to the SSD but the next problem seems to be that the formatted SDD drive seems to have no UUID.

As I'm kinda new to all this stuff it's hard to figure out the best way out of this situation. Is the M2.SSD not really suited for this kinda of usecase? Are there any steps I have to include when using the M2? 

Maybe you can point me in the right direction.

**Comment by szaimen** [comment]
Hi, I fear using an internal disk is indeed not possible currently.

However you could download this script onto your server: https://raw.githubusercontent.com/nextcloud/vm/master/not-supported/veracrypt-btrfs.sh and change this line: https://github.com/nextcloud/vm/blob/6556ef53ac6bae450bbeaa3b786fdaa5921d34a4/not-supported/veracrypt-btrfs.sh#L85 to e.g. `for drive in "/dev/nvme0n2"` or whatever your device is called.

**Reply by janivo** [reply]
Thanks for responding!

Yea, thats what I thought and exactly what I did. But it seems like that there are some other requirements/impediments for M2 drives as it errors in https://github.com/nextcloud/vm/blob/6556ef53ac6bae450bbeaa3b786fdaa5921d34a4/not-supported/veracrypt-btrfs.sh#L194 because the new partition created by veracrypt does not have a UUID.

I know that's out of scope of this guide but I thought that you might have an idea why that is.

**Reply by szaimen** [reply]
So it seems like the new partition does not have a PARTUUID? This is indeed very strange and no idea how to fix that :/
